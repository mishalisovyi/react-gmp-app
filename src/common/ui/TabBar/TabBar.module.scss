.TabBar {
  display: flex;
  column-gap: 32px;
}