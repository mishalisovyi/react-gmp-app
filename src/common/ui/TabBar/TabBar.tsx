import React, { memo, useCallback, useState } from 'react';
import memoize from 'fast-memoize';

import { Tab } from 'common/ui';
import styles from './TabBar.module.scss';

interface TabBarProps {
  tabsLabels: string[]
}

function TabBarComponent({ tabsLabels }: TabBarProps) {
  const [activeTabIndex, setActiveTabIndex] = useState(0);

  const handleTabClick = useCallback(memoize((tabIndex: number) => {
    return () => {
      setActiveTabIndex(tabIndex);
    };
  }), []);

  return (
    <div className={styles['TabBar']}>
      {tabsLabels.map((label, index) => {
        return (
          <Tab
            label={label}
            key={label}
            isActive={index === activeTabIndex}
            onClick={handleTabClick(index)}
          />
        );
      })}
    </div>
  );
}

export const TabBar = memo(TabBarComponent);
