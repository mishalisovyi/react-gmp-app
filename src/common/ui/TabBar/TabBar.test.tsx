import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { TabBar } from './TabBar';

const tabsData = [{
  title: 'Tab 1',
  value: 'tab_1',
}, {
  title: 'Tab 2',
  value: 'tab_2',
}];

const activeTabChangedHandler = jest.fn();

it('should render tab bar', () => {
  const tree = renderer.create(<TabBar tabs={tabsData} defaultActiveTab={tabsData[0].value} onActiveTabChanged={activeTabChangedHandler} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should change active tab', async () => {
  render(<TabBar tabs={tabsData} defaultActiveTab={tabsData[0].value} onActiveTabChanged={activeTabChangedHandler} />);

  await userEvent.click(screen.getByText(tabsData[1].title));

  expect(activeTabChangedHandler).toHaveBeenCalledWith(tabsData[1].value);
});
