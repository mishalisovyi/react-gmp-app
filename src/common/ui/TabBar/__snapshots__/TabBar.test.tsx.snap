// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render tab bar 1`] = `
<div>
  <div
    className="undefined undefined"
    onClick={[Function]}
  >
    Tab 1
  </div>
  <div
    className="undefined "
    onClick={[Function]}
  >
    Tab 2
  </div>
</div>
`;
